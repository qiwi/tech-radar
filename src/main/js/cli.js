#!/usr/bin/env node

import meow from 'meow'
import {generateTechRadar} from './index.js'

const cli = meow(
  `
    Usage:
      techradar  --csv-path /path/to/csv --out-dir /radar --versions 1.00 pathPrefix js
    Options
      --csvPath
      --outDir
      --version
      --pathPrefix
`,
  {
    importMeta: import.meta,
    flags: {
      csvPath: {
        type: 'string',
        isRequired: true,
      },
      outDir: {
        type: 'string',
        isRequired: true,
      },
      version: {
        type: 'string',
      },
      pathPrefix: {
        type: 'string',
      },
    },
  },
)

generateTechRadar(cli.flags)
