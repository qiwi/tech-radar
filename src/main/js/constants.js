import path from 'path'

export const langAndFw = 'languages-and-frameworks'
export const platforms = 'platforms'
export const tools = 'tools'
export const techniques = 'techniques'

export const tplDir = path.resolve('src/main/tpl')
export const tempDir = 'radar-temp'

export const radarSchema = {
  type: 'object',
  properties: {
    data: {
      type: 'array',
      items: {
        type: 'object',
        properties: {
          name: { type: 'string' },
          quadrant: { type: 'string' },
          ring: { type: 'string' },
          description: { type: ['string', 'null'] },
          moved: { type: ['string', 'integer', 'null'] },
        },
        required: ['name', 'quadrant', 'ring', 'description'],
      },
    },
    meta: {
      type: 'object',
      properties: {
        title: { type: 'string' },
        date: { type: 'string' },
        legend: { type: ['string', 'null'] },
      },
      required: ['title', 'date'],
    },
  },
}
export const settings = {
  svg_id: 'radar',
  width: 1450,
  height: 1100,
  colors: {
    background: '#fff',
    grid: '#bbb',
    inactive: '#ddd',
  },
  rings: [
    { name: 'ADOPT', color: '#93c47d', id: 'adopt' },
    { name: 'TRIAL', color: '#93d2c2', id: 'trial' },
    { name: 'ASSESS', color: '#fbdb84', id: 'assess' },
    { name: 'HOLD', color: '#efafa9', id: 'hold' },
  ],
  print_layout: true,
}
