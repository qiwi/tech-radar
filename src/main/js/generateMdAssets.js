import parse from 'csv-parse/lib/sync.js'
import fs from 'fs'
import fsExtra from 'fs-extra'
import path from 'path'

const tplPath = path.resolve('src/main/tpl')

export function generatePath({ name, quadrant, tempDirResolved }) {
  const entryMdName = name + '.md'
  return path.join(
    tempDirResolved,
    '/entries',
    quadrant.toLowerCase(),
    entryMdName,
  )
}

export function generateMd({ ring, description }) {
  return `---
ring: ${ring.toLowerCase()}
---
${description}`
}

export const generateMdAssets = ({ csvPath, tempDir }) => {
  const tempDirResolved = path.resolve(tempDir)
  const csvPathResolved = path.resolve(csvPath)
  fsExtra.copySync(tplPath, tempDirResolved)
  const radarData = fs.readFileSync(csvPathResolved)
  const records = parse(radarData, { columns: true })

  records.forEach(({ name, quadrant, ring, description }) => {
    const entryFilePath = generatePath({ name, quadrant, tempDirResolved })
    const content = generateMd({ ring, description })
    fs.writeFileSync(entryFilePath, content)
  })
}
