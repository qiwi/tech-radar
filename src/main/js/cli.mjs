#!/usr/bin/env node

import meow from 'meow'
import {run} from './index.js'

export const cli = meow(
  `
    Usage:
      techradar  --input /path/to/csv --output /radar --basePrefix tech-radar --autoscope true --nav-page true --nav-title title --nav-footer footer
    Options
      --input
      --output
      --basePrefix
      --autoscope
      --nav-page
      --nav-title
      --nav-footer
      --temp
`,
  {
    importMeta: import.meta,
    flags: {
      input: {
        type: 'string',
        isRequired: true,
      },
      output: {
        type: 'string',
        isRequired: true,
      },
      basePrefix: {
        type: 'string',
      },
      cwd: {
        type: 'string',
      },
      autoscope: {
        type: 'boolean'
      },
      navPage: {
        type: 'boolean'
      },
      navTitle: {
        type: 'string'
      },
      navFooter: {
        type: 'string'
      },
      temp: {
        type: 'string'
      }
    },
  },
)

run(cli.flags)
