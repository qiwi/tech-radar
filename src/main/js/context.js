import { uniq } from 'lodash-es'
import path from 'path'

import { read } from './reader.js'
import { getDirs } from './util.js'

/**
 * add param file to context
 * @param sources
 * @returns contexts
 */
export const init = (sources) => sources.map((s) => ({ file: s }))
/**
 * add param data to context
 * @param contexts
 * @returns contexts
 */
export const readFiles = (contexts) =>
  contexts.map((c) => ({
    ...c,
    data: read(c.file),
  }))
/**
 * add param base to context
 * @param contexts
 * @returns contexts
 */
export const resolveBases = (contexts) => {
  const dates = contexts.map(({ data }) => data.meta.date)
  if (contexts.length === 1)
    return [
      {
        ...contexts[0],
        date: contexts[0].data.meta.date,
        base: contexts[0].data.meta.date,
      },
    ]
  const filesWithNoExt = contexts.map(({ file }, i) =>
    path.join(file.slice(0, -path.basename(file).length), dates[i]),
  )

  if (uniq(filesWithNoExt).length !== filesWithNoExt.length) {
    throw new Error('same dates in one directory')
  }
  return getDirs(filesWithNoExt).map((base, i) => ({
    ...contexts[i],
    date: dates[i],
    base,
  }))
}
/**
 * sort contexts with path and date
 * @param contexts
 * @returns contexts
 */
export const sortContexts = (contexts) => {
  if (contexts.length === 1) return contexts
  return contexts.sort((a, b) => {
    if (path.dirname(a.file) > path.dirname(b.file)) return 1
    if (path.dirname(a.file) < path.dirname(b.file)) return -1
    if (a.date > b.date) return 1
    if (a.date < b.date) return -1
    return 0
  })
}
