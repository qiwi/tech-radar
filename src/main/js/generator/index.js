import Eleventy from '@11ty/eleventy'
import fse from 'fs-extra'
import { uniq } from 'lodash-es'
import path, { dirname } from 'path'
import { fileURLToPath } from 'url'

import {
  defNavFooter,
  defNavTitle,
  radarSchema, settings as defaultSettings,
  tplNavPage,
} from '../constants.js'
import { genMdAssets } from './markdown.js'
import { sortContextsByDate, writeSettings } from '../util.js'
import { validate } from '../validator.js'

const __dirname = dirname(fileURLToPath(import.meta.url))
const tplDir = path.resolve(__dirname, '../../tpl')

/**
 * generate static sites from array radarDocument
 * @param contexts
 * @param _output
 * @param basePrefix
 */
export const generateStatics = async (contexts, _output, basePrefix, temp) => {
  return

  const statics = await contexts.reduce(async (_r, context) => {
    const _m = await _r
    const { data, base } = context
    if (!validate(data, radarSchema) || Object.keys(data).length === 0)
      return context

    const output = base ? path.join(_output, base) : _output
    const pathPrefix = basePrefix ? basePrefix + '/' + base : undefined
    Object.assign(context, {
      date: data.meta.date,
      title: data.meta.title,
      temp,
      output,
      pathPrefix
    })

    try {
      genMdAssets(context)
      writeSettings(context)
      await genEleventy(context)
    } catch (err) {
      console.error('genStatics', err)
    }
    return [..._m, output]
  }, [])
  console.log('statics=', statics)
  return contexts.map((context) => ({
    ...context,
    prefix: basePrefix ? basePrefix + '/' + context.base : context.base,
  }))
}

export const genConfig = async ({temp, output, title, prefix, date}) => {
  const configExtPath = path.resolve(__dirname, '.eleventy.cjs')
  const configMixin = {extra: {temp, title, prefix, date, output }}
  const configPath = path.join(temp, 'config.js')
  const configContents = `
module.exports = (config) => require('${configExtPath}')(Object.assign(config, ${JSON.stringify(configMixin)}))
`
  await fse.writeFile(configPath, configContents, 'utf8')

  return configPath
}

export const genSettings = async ({temp, document, title, prefix, date, output}) => {
  const quadrants = [
    { name: document.quadrantTitles.q1 || 'Q1', id: 'q1' },
    { name: document.quadrantTitles.q2 || 'Q2', id: 'q2' },
    { name: document.quadrantTitles.q3 || 'Q3', id: 'q3' },
    { name: document.quadrantTitles.q4 || 'Q4', id: 'q4' },
  ]

  const extra = { output, title, prefix, temp, date }
  const settins = {...defaultSettings, extra, quadrants}
  const settingsPath = path.join(temp, '_data/settins.json')

  await fse.writeFile(settingsPath, JSON.stringify(settins))
}

/**
 * generate static site with using 11ty
 * @param radar
 */
export const genEleventy = async (radar) => {
  await fse.copy(tplDir, radar.temp)

  const configPath = await genConfig(radar)
  await genMdAssets(radar)
  await genSettings(radar)

  const elev = new Eleventy(radar.temp, radar.output, { configPath })

  await elev.init()
  await elev.write()
}

export const genStatics = async(radar) => {
  await fse.copy(tplDir, radar.temp)
  await genEleventy(radar)
}

export const genRedirects = async({radars, output}) => {
  const redirectTpl = await fse.readFile(path.join(tplDir, 'redirect-page/index.html'), 'utf8')

  await Promise.all(Object.entries(radars.reduce((m, {scope, date}) => {
    const prev = m[scope]

    if (!prev || prev < date) {
      m[scope] = date
    }
    return m
  }, {})).map(([scope, date]) =>
    fse.writeFile(path.join(output, scope, 'index.html'), redirectTpl.replace('###', date))
  ))
}

export const genNavPage = async ({
 radars,
 output,
 navPage,
 navTitle,
 navFooter,
}) => {
  if (!navPage) return
  if (!navTitle) navTitle = defNavTitle
  if (!navFooter) navFooter = defNavFooter

  const headers = uniq(radars.map(r => r.scope))
  const navBlock = headers
    .map((_scope) => `<div class="tile">
<h2>${_scope}</h2>
<ul>
  ${
    radars
      .filter(({scope}) => scope === _scope)
      .map(({prefix, date}) => `<li><a class="link" href=${prefix}> ${date}</a></li>`)
      .join('\n')
  }
</ul>
</div>`)
    .join('\n')


  // Main nav page
  await fse.copy(path.join(tplDir, 'nav-page'), output)
  const tplHtml = await fse.readFile(path.join(output, 'index.html'), 'utf8') // eslint-disable-line sonarjs/no-duplicate-string
  const html = tplHtml
    .replace('#nav_page-content', navBlock)
    .replace('#nav_page-title', navTitle)
    .replace('#nav_page-footer', navFooter)

  await fse.writeFile(path.join(output, 'index.html'), html)

  // Latest scoped radars aliases
  // const redirect = fse.readFileSync(path.resolve(__dirname, '../redirect-page/index.html'), 'utf8')
  // Object.entries(contexts.reduce((m, {base, date}) => {
  //   const scope = path.dirname(base)
  //   const prev = m[scope]
  //
  //   if (!prev || prev < date) {
  //     m[scope] = date
  //   }
  //   return m
  // }, {})).forEach(([scope, date]) => {
  //   fse.writeFileSync(path.join(output, scope, 'index.html'), redirect.replace('###', date))
  // })
}
