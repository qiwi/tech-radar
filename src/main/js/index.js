import fse from 'fs-extra'
import path from 'node:path'

import { tplDir } from './constants.js'
import { genNavPage, genRadars, genRedirects } from './generator/index.js'
import { getSources, parse } from './parser/index.js'
import { getDirs, tempDir } from './util.js'

/**
 * @description
 * Generate static sites from csv/json/yml radar declarations
 *
 * @func
 * @param {Object} options
 * @param {string} options.input globby pattern for input files
 * @param {string} options.output output directory
 * @param {string} options.cwd current working directory
 * @param {string?} options.basePrefix web app root level prefix
 * @param {boolean?} options.autoscope consider same-scoped files as subversions of a single radar
 * @param {boolean?} options.navPage Generate navigation page
 * @param {string?} options.navTitle Nav page title
 * @param {string?} options.navFooter Nav page footer
 * @param {string?} options.temp Temp directory
 *
 * @return {Promise<void>}
 */
export const run = async (options) => {
  const ctx = await getContext(options)

  return readSources(ctx)
    .then(parseRadars)
    .then(sortRadars)
    .then(resolveMoves)
    .then(renderRadars)
    .finally(() => cleanTemp(ctx))
}

const getContext = async ({
  input,
  output,
  cwd = process.cwd(),
  basePrefix = '/',
  autoscope = false,
  navPage = false,
  navTitle,
  navFooter,
  temp,
  templates,
  renderSettings,
} = {}) => {
  const ctx = {
    input,
    output: path.resolve(cwd, output),
    cwd,
    basePrefix,
    autoscope,
    navPage,
    navTitle,
    navFooter,
    temp: temp || (await tempDir()),
    templates,
    renderSettings,
  }

  ctx.ctx = ctx // context self-ref to simplify pipelining

  return ctx
}

const readSources = async ({ ctx, cwd, input }) => {
  ctx.sources = await getSources(input, cwd)
  ctx.scopes = getDirs(ctx.sources).map(path.dirname)
  return ctx
}

const parseRadars = async ({ ctx, sources, scopes }) => {
  ctx.radars = await Promise.all(
    sources.map(async (file, i) => {
      const document = await parse(file)

      return {
        document,
        source: file,
        scope: scopes[i],
        date: document.meta.date,
        title: document.meta.title,
      }
    }),
  )

  return ctx
}

const renderRadars = async ({ ctx, output }) => {
  await genRadars(ctx)
  await genNavPage(ctx)
  await genRedirects(ctx)
  await fse.copy(path.join(tplDir, 'assets'), output) // shared static assets

  // console.log('radars', radars)
  // console.log('radar', JSON.stringify(radars[3], null, 2))
  return ctx
}

const resolveMoves = async ({ ctx, radars, autoscope }) => {
  if (!autoscope) {
    return ctx
  }

  const rings = {
    hold: 0,
    assess: 1,
    trial: 2,
    adopt: 3,
  }

  const getRingWeight = (ring) => rings[ring.toLowerCase()]

  radars.forEach(({ document: { data }, scope }, i) => {
    data.forEach((entry) => {
      const { name, ring, moved } = entry
      const lowerName = name.toLowerCase()
      const prevRadar = radars[i + 1] // NOTE sorted by desc date
      const prevEntry =
        prevRadar &&
        prevRadar.scope === scope &&
        prevRadar.document.data.find(
          ({ name: _name }) => _name.toLowerCase() === lowerName,
        )

      entry.moved = prevEntry
        ? Math.sign(getRingWeight(ring) - getRingWeight(prevEntry.ring))
        : moved || 0
    })
  })

  return ctx
}

const sortRadars = async ({ ctx, radars }) => {
  radars.sort((a, b) => {
    if (path.dirname(a.source) > path.dirname(b.source)) return 1
    if (path.dirname(a.source) < path.dirname(b.source)) return -1

    return Math.sign(Date.parse(b.date) - Date.parse(a.date))
  })

  return ctx
}

const cleanTemp = async ({ ctx, temp }) => {
  await fse.remove(temp)
  return ctx
}
