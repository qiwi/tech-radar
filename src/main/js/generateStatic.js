import Eleventy from '@11ty/eleventy'
import fs from 'fs'
import fsExtra from 'fs-extra'
import { uniq } from 'lodash-es'
import path, { dirname } from 'path'
import { fileURLToPath } from 'url'

import {
  defNavFooter,
  defNavTitle,
  radarSchema,
  tempDir,
  tplNavPage,
} from './constants.js'
import { genMdAssets } from './generateMdAssets.js'
import { sortContextsByDate, writeSettings } from './util.js'
import { validate } from './validator.js'

const __dirname = dirname(fileURLToPath(import.meta.url))

/**
 * generate static sites from array radarDocument
 * @param contexts
 * @param _output
 * @param basePrefix
 */
export const generateStatics = async (contexts, _output, basePrefix) => {
  const statics = await contexts.reduce(async (_r, context) => {
    const _m = await _r
    const { data, base } = context
    if (!validate(data, radarSchema) || Object.keys(data).length === 0)
      return context

    const temp = tempDir
    const output = base ? path.join(_output, base) : _output
    const pathPrefix = basePrefix ? basePrefix + '/' + base : undefined

    global._11ty_ = {
      date: data.meta.date,
      title: data.meta.title,
      output,
      temp,
      pathPrefix,
    }

    try {
      genMdAssets(data, temp)
      writeSettings(data, temp)
      await genEleventy(temp, output)
    } catch (err) {
      console.error('genStatics', err)
    }
    return [..._m, output]
  }, [])
  console.log('statics=', statics)
  return contexts.map((context) => ({
    ...context,
    prefix: basePrefix ? basePrefix + '/' + context.base : context.base,
  }))
}

/**
 * generate static site with using 11ty
 * @param temp
 * @param output
 */
export const genEleventy = async (temp, output) => {
  const elev = new Eleventy(temp, output)
  elev.setConfigPathOverride('src/main/js/11ty/.eleventy.cjs')
  await elev.init()
  await elev.write()
  fsExtra.removeSync(temp)
}

export const genNavigationPage = (
  contexts,
  output,
  navPage,
  input,
  navTitle,
  navFooter,
) => {
  if (!navPage) return
  if (!navTitle) navTitle = defNavTitle
  if (!navFooter) navFooter = defNavFooter

  const dirs = contexts.map(({ file }) => path.dirname(file))
  const links = uniq(dirs.map((dir) => dir.split('/').pop())).reduce(
    (r, dir) => ({ ...r, [dir]: [] }),
    {},
  )

  contexts.sort(sortContextsByDate).forEach((context) => {
    const keys = Object.keys(links)
    const key =
      keys.length === 1
        ? keys[0]
        : keys.find((dir) => dir === context.base.split('/')[0])
    if (!key) return
    const link = `<a class="link" href=${context.base}> ${context.data.meta.date} </a>`
    links[key].push(link)
  })

  const li = (link) => `<li>${link}</li>`
  const contentPage = Object.keys(links).map(
    (key) => `<div class="tile">
<h2>${key}</h2>
<ul>
  ${links[key].map(li).join('\n ')}
</ul>
</div>`,
  )

  // Main nav page
  fsExtra.copySync(tplNavPage, output)
  const tplHtml = fs.readFileSync(path.join(output, 'index.html'), 'utf8') // eslint-disable-line sonarjs/no-duplicate-string
  const html = tplHtml
    .replace('#nav_page-content', contentPage.join('\n'))
    .replace('#nav_page-title', navTitle)
    .replace('#nav_page-footer', navFooter)
  fs.writeFileSync(path.join(output, 'index.html'), html)

  // Latest scoped radars aliases
  const redirect = fs.readFileSync(path.resolve(__dirname, '../redirect-page/index.html'), 'utf8')
  Object.entries(contexts.reduce((m, {base, date}) => {
    const scope = path.dirname(base)
    const prev = m[scope]

    if (!prev || prev < date) {
      m[scope] = date
    }
    return m
  }, {})).forEach(([scope, date]) => {
    fs.writeFileSync(path.join(output, scope, 'index.html'), redirect.replace('###', date))
  })
}
