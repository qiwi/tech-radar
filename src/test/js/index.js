import fs from 'fs'
import fsExtra from 'fs-extra'
import path from 'path'

import {
  generateMd,
  generateMdAssets,
  generatePath,
} from '../../main/js/generateMdAssets.js'
import { generateStatics } from '../../main/js/index.js'

describe('generate md assets', () => {
  it('files write check', () => {
    const csvPath = path.join(__dirname, '../stub/test.csv')
    const outDir = path.join(__dirname, 'temp')
    generateMdAssets({ csvPath, tempDir: outDir })

    const tsMdData = fs.readFileSync(
      path.join(outDir, 'entries', 'languages-and-frameworks', 'TypeScript.md'),
      'utf8',
    )
    const nodeMdData = fs.readFileSync(
      path.join(outDir, 'entries', 'platforms', 'Nodejs.md'),
      'utf8',
    )
    const hexMdData = fs.readFileSync(
      path.join(
        outDir,
        'entries',
        'techniques',
        'Гексагональная архитектура.md',
      ),
      'utf8',
    )
    const codMdData = fs.readFileSync(
      path.join(outDir, 'entries', 'tools', 'codeclimate.md'),
      'utf8',
    )

    expect({ tsMdData, nodeMdData, hexMdData, codMdData }).toMatchSnapshot()
  })
  afterAll(() => {
    fsExtra.removeSync(path.join(__dirname, 'temp'))
  })

  it('generateMd ', () => {
    const contentMd = `---
ring: hold
---
Мидвары поверх http-server`
    expect(
      generateMd({ ring: 'Hold', description: 'Мидвары поверх http-server' }),
    ).toBe(contentMd)
  })

  it('generatePath ', () => {
    expect(
      generatePath({
        name: 'Redux',
        quadrant: 'languages-and-frameworks',
        tempDirResolved: 'test',
      }),
    ).toBe('test/entries/languages-and-frameworks/Redux.md')
  })
})

describe('generate e11y app', () => {
  it('', async () => {
    const csvPath = path.join(__dirname, '../stub/test.csv')
    const outDir = path.resolve('temp')
    generateMdAssets({ csvPath, tempDir: outDir })
    global.tempDir = 'temp'
    global.outDir = 'dist'
    await generateStatics(global.tempDir, global.outDir)

    const getFileStruct = (dir, result = []) => {
      fs.readdirSync(dir).forEach((elem) => {
        const elemPath = dir + '/' + elem
        const stat = fs.statSync(elemPath)
        if (stat.isDirectory()) {
          result = [...getFileStruct(elemPath, result)]
        } else {
          result.push(elemPath)
        }
      })
      return result
    }
    const fileStruct = getFileStruct(path.resolve('dist'))
    const normalizedFileStruct = fileStruct.map((el) => /dist(.+)/.exec(el)[1])
    expect(normalizedFileStruct).toMatchSnapshot()
  })
  afterAll(() => {
    fsExtra.removeSync(path.resolve('dist'))
    fsExtra.removeSync(path.resolve('temp'))
  })
})
