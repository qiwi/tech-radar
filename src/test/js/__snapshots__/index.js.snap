// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate e11y app  1`] = `
Array [
  "/entries/languages-and-frameworks/TypeScript/index.html",
  "/entries/platforms/Nodejs/index.html",
  "/entries/techniques/Гексагональная архитектура/index.html",
  "/entries/tools/codeclimate/index.html",
  "/favicon.ico",
  "/index.html",
  "/radar.css",
  "/radar.js",
]
`;

exports[`generate md assets files write check 1`] = `
Object {
  "codMdData": "---
ring: trial
---
Статический анализатор кода",
  "hexMdData": "---
ring: assess
---
Унификации контракта интерфейсов различных слоев приложений",
  "nodeMdData": "---
ring: adopt
---
",
  "tsMdData": "---
ring: adopt
---
Статически типизированный ЖС",
}
`;
